gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/ProjectInfo.java

```diff
@@ -0,0 +1,310 @@
+      this.parent = parent;
+      return this;
+    }
+
+    public ProjectInfo build() {
+      return new ProjectInfo(this);
+    }
+  }
+
+  ProjectInfo(Builder builder) {
+    this.name = builder.name;
+    this.id = checkNotNull(builder.id);
+    this.labels = ImmutableMap.copyOf(builder.labels);
+    this.number = builder.number;
+    this.state = builder.state;
+    this.createTimeMillis = builder.createTimeMillis;
+    this.parent = builder.parent;
+  }
+
+  /**
+   * Get the unique, user-assigned ID of the project.
+   *
+   * This field cannot be changed after the server creates the project.
+   */
+  public String id() {
+    return id;
+  }
+
+  /**
+   * Get the user-assigned name of the project.
+   *
+   * This field is optional, can remain unset, and can be changed after project creation.
+   */
+  public String name() {
+    return name;
+  }
+
+  /**
+   * Get number uniquely identifying the project.
+   *
+   * This field is set by the server and is read-only.
+   */
+  public Long number() {
+    return number;
+  }
+
+  /**
+   * Get the immutable map of labels associated with this project.
+   */
+  public Map<String, String> labels() {
+    return labels;
+  }
+
+  /**
+   * Get the project's lifecycle state.
+   *
+   * This is a read-only field. To change the lifecycle state of your project, use the
+   * {@code delete} or {@code undelete} method.
+   */
+  public State state() {
+    return state;
+  }
+
+  /**
+   * Get the project's creation time (in milliseconds).
+   *
+   * This field is set by the server and is read-only.
+   */
+  public Long createTimeMillis() {
+    return createTimeMillis;
+  }
+
+  /**
+   * Get the parent of the project.
+   *
+   * The parent cannot be changed after the server creates the project.
+   */
+  public ResourceId parent() {
+    return parent;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    return obj instanceof ProjectInfo && Objects.equals(toPb(), ((ProjectInfo) obj).toPb());
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(name, id, labels, number, state, createTimeMillis, parent);
+  }
+
+  public static Builder builder(String id) {
+    return new Builder().id(id);
+  }
+
+  public Builder toBuilder() {
+    return new Builder()
+        .name(name)
+        .id(id)
+        .labels(labels)
+        .number(number)
+        .state(state)
+        .createTimeMillis(createTimeMillis)
+        .parent(parent);
+  }
+
+  com.google.api.services.cloudresourcemanager.model.Project toPb() {
+    com.google.api.services.cloudresourcemanager.model.Project projectPb =
+        new com.google.api.services.cloudresourcemanager.model.Project();
+    projectPb.setName(name);
+    projectPb.setProjectId(id);
+    projectPb.setLabels(labels);
+    projectPb.setProjectNumber(number);
+    if (state != null) {
+      projectPb.setLifecycleState(state.toString());
+    }
+    if (createTimeMillis != null) {
```

Comment:
I checked using the API Explorer that it doesn't complain and it doesn't update read-only values.  However, since it's not documented explicitly that the service won't complain, I can take out the read-only fields (number, createTimeMillis, and lifecycleState).